python/cuml/tsa/arima.pyx

```diff
@@ -880,58 +919,57 @@ class ARIMA(Base):
         p, q, P, Q, k = (order.p, order.q, order.P, order.Q, order.k)
         N = self.complexity
 
-        if type(x) is list or x.shape != (N, self.batch_size):
-            x_mat = np.reshape(x, (N, self.batch_size), order='F')
-        else:
-            x_mat = x
+        x_mat = (np.reshape(x, (N, self.batch_size), order='F')
+                 .astype(np.float64))
 
-        params = dict()
         # Note: going through np.array to avoid getting incorrect strides when
-        # batch_size is 1
+        # one dimension is 1
         if k > 0:
-            params["mu"] = np.array(x_mat[0], order='F')
+            mu_temp = np.array(x_mat[0], order='F')
+            self._mu_, *_ = input_to_cuml_array(mu_temp)
         if p > 0:
-            params["ar"] = np.array(x_mat[k:k+p], order='F')
+            ar_temp = np.array(x_mat[k:k+p], order='F')
+            self._ar_, *_ = input_to_cuml_array(ar_temp)
         if q > 0:
-            params["ma"] = np.array(x_mat[k+p:k+p+q], order='F')
+            ma_temp = np.array(x_mat[k+p:k+p+q], order='F')
+            self._ma_, *_ = input_to_cuml_array(ma_temp)
         if P > 0:
-            params["sar"] = np.array(x_mat[k+p+q:k+p+q+P], order='F')
+            sar_temp = np.array(x_mat[k+p+q:k+p+q+P], order='F')
+            self._sar_, *_ = input_to_cuml_array(sar_temp)
         if Q > 0:
-            params["sma"] = np.array(x_mat[k+p+q+P:k+p+q+P+Q], order='F')
-        params["sigma2"] = np.array(x_mat[k+p+q+P+Q], order='F')
-
-        self.set_params(params)
+            sma_temp = np.array(x_mat[k+p+q+P:k+p+q+P+Q], order='F')
+            self._sma_, *_ = input_to_cuml_array(sma_temp)
+        sigma2_temp = np.array(x_mat[k+p+q+P+Q], order='F')
+        self._sigma2_, *_ = input_to_cuml_array(sigma2_temp)
```

Comment:
For reference, `input_to_cuml_array` can convert the order and dtype in a single step and `CumlArray` is capable of slicing. So you can reduce much of this code. For example, `x_mat` can be created with:
```python
x_mat = input_to_cuml_array(np.reshape(x, (N, self.batch_size)), order='F', convert_to_dtype=np.float64).array
```
Then only 1 data transfer happens to the device and all slicing happens on the GPU
Note: Adding `.array` to the end of `input_to_cuml_array` eliminates the need to do `x_mat, _, *_ = input_to_cuml_array()`. And then you can just create the individual attributes with slicing like normal:
```python
self._mu_ = x_mat[0]
self._ar_ = x_mat[k:k+p]
...
```

Reply:
Slicing doesn't create a contiguous array, it just returns the same array with different offsets and strides.
I went for another solution: interface the CUDA versions of `pack` and `unpack` (so only one host <-> device copy).

Reply:
Sounds good to me.